testing/camunda-process-test-java/src/main/java/io/camunda/process/test/api/CamundaProcessTestExtension.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.process.test.api;
+
+import static org.junit.platform.commons.util.ReflectionUtils.makeAccessible;
+
+import io.camunda.process.test.impl.extension.CamundaProcessTestContextImpl;
+import io.camunda.process.test.impl.runtime.CamundaContainerRuntime;
+import io.camunda.process.test.impl.runtime.CamundaContainerRuntimeBuilder;
+import io.camunda.zeebe.client.ZeebeClient;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+import org.junit.jupiter.api.extension.ExtensionContext.Store;
+import org.junit.platform.commons.util.ExceptionUtils;
+import org.junit.platform.commons.util.ReflectionUtils;
+
+/**
+ * A JUnit extension that provides the runtime for process tests.
+ *
+ * <p>Before each test method:
+ *
+ * <ul>
+ *   <li>Start the runtime
+ *   <li>Inject a {@link ZeebeClient} to a field in the test class
+ *   <li>Inject a {@link CamundaProcessTestContext} to a field in the test class
+ * </ul>
+ *
+ * <p>After each test method:
+ *
+ * <ul>
+ *   <li>Close created {@link ZeebeClient}s
+ *   <li>Stop the runtime
+ * </ul>
+ */
+public class CamundaProcessTestExtension implements BeforeEachCallback, AfterEachCallback {
+
+  /** The JUnit extension namespace to store the runtime and context. */
+  public static final Namespace NAMESPACE = Namespace.create(CamundaProcessTestExtension.class);
+
+  /** The JUnit extension store key of the runtime. */
+  public static final String STORE_KEY_RUNTIME = "camunda-process-test-runtime";
+
+  /** The JUnit extension store key of the context. */
+  public static final String STORE_KEY_CONTEXT = "camunda-process-test-context";
+
+  private final List<ZeebeClient> createdClients = new ArrayList<>();
```

Comment:
❓ Why a list of clients ? I expect a single client to be inizialized

Reply:
In most cases, it might be a single client. However, I want to be more flexible.